.github/workflows/dependabot_changelog.yml

```diff
@@ -0,0 +1,23 @@
+name: Write changelog for dependabot PR
+on:
+  pull_request:
+    types:
+      - opened
+
+jobs:
+  add-changelog:
+    runs-on: 'ubuntu-latest'
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
```

Comment:
To the reviewer: please double check that
- this correctly checks out the branch that the PR is tracking
- so that `git push` makes a new commit and updates the branch to point to that new commit

I think this is correct, but I am not an expert with git's plumbing.

Reply:
looks right to me.

(I think, by default, it checks out the SHA of a merge between the PR branch and the target branch?)